Cargo.toml

```diff
@@ -13,24 +13,25 @@ keywords    = ["apns", "apple", "push", "async", "http2"]
 repository  = "https://github.com/walletconnect/a2.git"
 homepage    = "https://github.com/walletconnect/a2"
 documentation = "https://docs.rs/a2"
-edition = "2018"
+edition = "2021"
+
+[features]
+tracing = ["dep:tracing"]
 
 [dependencies]
-serde = "1"
+serde = { version = "1", features = ["derive"] }
 erased-serde = "0.3"
-serde_derive = "1"
 serde_json = "1"
 thiserror = "1"
 openssl = "0.10"
-futures = "0.3"
+hyper = { version = "0.14", features = ["client", "http2"] }
+hyper-alpn = "0.4"
 http = "0.2"
 base64 = "0.13"
-log = "0.4"
-hyper = { version = "0.14", features = ["client", "http2", "tcp"] }
-hyper-alpn = "0.3"
+tracing = { version = "0.1.36", optional = true }
 
 [dev-dependencies]
 argparse = "0.2"
-pretty_env_logger = "0.4"
-indoc = "1"
+tracing-subscriber = "0.3.15"
 tokio = { version = "1", features = ["rt-multi-thread", "macros"] }
+hyper = { version = "0.14", features = ["client", "http2", "tcp"] }
```

Comment:
why is this only a dev dependency now?

Reply:
it's both but we only need the `tcp` feature in dev for the examples

Reply:
You might want to use the new resolver. The old one will use the largest set of features for both builds (a cargo bug).

Reply:
We have moved to `edition = 2021` which implies `resolver = 2` according to docs